blue_controllers:

  ### Controller for publishing to the joint_states topic

  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 125

  ### Gripper controller

  gripper_controller:
    type: "gripper_action_controller/GripperActionController"
    joints: gripper_joint
    goal_tolerance: 0.01
    stalled_velocity_threshold: 0.01
    stall_timeout: 0.2
    gains:
      gripper_joint: {p: 20, i: 5, d: 1, i_clamp: 1}

  ### Joint controllers

  torque_controller:
    type: effort_controllers/JointGroupEffortController
    joints:
      - lift_joint
      - roll_joint

  joint_position_controller:
    type: effort_controllers/JointGroupPositionController
    joints:
      - lift_joint
      - roll_joint
    lift_joint:
      pid: {p: 10, i: 0, d: 0.5, i_clamp: 0}
    roll_joint:
      pid: {p: 10, i: 0, d: 0.5, i_clamp: 0}
